### PR TITLE
feat: allow on_change callback in auto-polling mode

### LINKED ISSUE #9: Call a callback when the configuration has changed
Allow a client to specify a callback that should be called when automatic polling fetches an updated configuration. The callback shouldn't be called if the configuration hasn't been changed.

### PR DESCRIPTION
**Pull request by randycoulman**
Closes #9 

The callback is called whenever the configuration changes, either by polling or a forced-refresh.

In order to test the on_changed callback feature, we need to use a real cache instead of a mock, because there doesn't seem to be a good way to mock the behavior we need correctly.

### CODE REVIEW

**Comment by randycoulman** [comment]
I'm going to add some exception handling around the callback before I merge this; will update the PR when I get a chance.

**Comment by randycoulman** [comment]
I've now added the exception handling.

**Review by igorescobar — APPROVED** [review]
Looking good! Sorry about the delay! Today was wild over here :(

**Pull request merged**
